src/allocator.cpp

```diff
@@ -104,11 +105,20 @@ void* PoolAllocator::fastMalloc(size_t size)
     d->budgets_lock.lock();
 
     // find free budget
-    std::list<std::pair<size_t, void*> >::iterator it = d->budgets.begin();
+    std::list<std::pair<size_t, void*> >::iterator it = d->budgets.begin(), it_max = d->budgets.end(), it_min = d->budgets.end();
```

Comment:
Why are these max/min iterators initialized to end() instead of begin() ?
If the budgets are empty, `if (d->budgets.size() >= d->size_threshold)` will always be false.